app/features/build/build_controller.rb

```diff
@@ -21,17 +21,28 @@ class BuildController < AuthenticatedControllerBase
         build_number: build_number
       )
 
-      if current_build_runner.nil?
-        # Loading the output of a finished build after a server restart isn't finished yet
-        # see https://github.com/fastlane/ci/issues/312 for more information
-        raise "Couldn't find build runner for project #{project_id} with build_number #{build_number}"
+      if current_build_runner
+        existing_rows = current_build_runner.all_build_output_log_rows.map(&:html).join("\n")
+      else
+        # `current_build_runner` is only defined if the build was just run a while back
+        # if the server was restarted, we're gonna end here in this code block
+        build_log_artifact = build.artifacts.find do |current_artifact|
+          # We can improve the detection in the future, to actually mark an artifact as "default output"
+          current_artifact.type == "log" && current_artifact.reference.end_with?("fastlane.log")
+        end
+
+        if build_log_artifact
+          existing_rows = File.read(build_log_artifact.reference).gsub("\n", "<br />")
```

Comment:
Please use https://github.com/fastlane/ci/blob/61e42317bc8e3c3104406368b517e8691f4ed7b6/app/shared/models/artifact_provider.rb#L26 instead of `reference`

Reply:
Ah nice, will update tomorrow 👍